x-pack/plugins/observability/public/hooks/slo/use_create_slo.ts

```diff
@@ -48,24 +48,20 @@ export function useCreateSlo() {
         // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
         await queryClient.cancelQueries(sloKeys.lists());
 
-        const latestFetchSloListRequest = (
-          queryClient.getQueriesData<FindSLOResponse>(sloKeys.lists()) || []
+        const latestQueriesData = (
+          queryClient.getQueriesData<FindSLOResponse>(sloKeys.lists()) ?? []
         ).at(0);
 
-        const [queryKey, data] = latestFetchSloListRequest || [];
+        const [queryKey, data] = latestQueriesData || [];
 
         const newItem = { ...slo, id: uuidv1() };
-
         const optimisticUpdate = {
           ...data,
-          results: [...(data?.results || []), { ...newItem }],
+          results: [...(data?.results ?? []), newItem],
           total: data?.total ? data.total + 1 : 1,
         };
 
-        // Optimistically update to the new value
-        if (queryKey) {
-          queryClient.setQueryData(queryKey, optimisticUpdate);
-        }
```

Comment:
By doing this, even if there was no entry in the queries cache, we add the new slo into the list page.